fili-core/src/main/java/com/yahoo/bard/webservice/table/PermissiveConcretePhysicalTable.java

```diff
@@ -13,38 +14,73 @@
- * {@link BasePhysicalTable#getAllAvailableIntervals()}, and {@link BasePhysicalTable#getTableAlignment()}.
+ * This is different from {@link ConcretePhysicalTable}. {@link PermissiveConcretePhysicalTable} is backed by
+ * {@link PermissiveAvailability}. The different Availability affects how available intervals of the table are
+ * calculated and returned.
  */
-public class PermissiveConcretePhysicalTable extends ConcretePhysicalTable {
+public class PermissiveConcretePhysicalTable extends BasePhysicalTable {
+
+    private final DataSourceName dataSourceName;
+
     /**
      * Create a permissive concrete physical table.
      *
      * @param name  Name of the physical table as TableName
      * @param timeGrain  time grain of the table
      * @param columns  The columns for this table
      * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
-     * @param dataSourceMetadataService  Data source metadata service containing availability data for the table
+     * @param metadataService  Data source metadata service containing availability data for the table
      */
     public PermissiveConcretePhysicalTable(
             @NotNull TableName name,
             @NotNull ZonedTimeGrain timeGrain,
             @NotNull Set<Column> columns,
             @NotNull Map<String, String> logicalToPhysicalColumnNames,
-            @NotNull DataSourceMetadataService dataSourceMetadataService
+            @NotNull DataSourceMetadataService metadataService
+    ) {
+        this(
+                name,
+                timeGrain,
+                columns,
+                logicalToPhysicalColumnNames,
+                new PermissiveAvailability(DataSourceName.of(name.asName()), metadataService)
+        );
+    }
+
+    /**
+     * Create a permissive concrete physical table.
+     *
+     * @param name  Name of the physical table as TableName
+     * @param timeGrain  time grain of the table
+     * @param columns  The columns for this table
+     * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
+     * @param availability  Availability that serves interval availability for columns
+     */
+    public PermissiveConcretePhysicalTable(
+            @NotNull TableName name,
+            @NotNull ZonedTimeGrain timeGrain,
+            @NotNull Set<Column> columns,
+            @NotNull Map<String, String> logicalToPhysicalColumnNames,
+            @NotNull PermissiveAvailability availability
     ) {
         super(
                 name,
                 timeGrain,
                 columns,
                 logicalToPhysicalColumnNames,
-                new PermissiveAvailability(name, dataSourceMetadataService)
+                availability
         );
+        this.dataSourceName = availability.getDataSourceName();
+    }
+
+    public DataSourceName getDataSourceName() {
+        return dataSourceName;
+    }
```

Comment:
This redundancy strikes me as wrong.  Go ahead and pull the 'SingleDataSourcePhysicalTable' logic up into a parent class and simplify Permissive and Strict subclasses.